src/sentences/bww.rs

```diff
@@ -0,0 +1,138 @@
+/// ```
+/// Field Number:
+/// 1. Bearing, degrees True
+/// 2. T = True
+/// 3. Bearing Degrees, Magnetic
+/// 4. M = Magnetic
+/// 5. TO Waypoint ID
+/// 6. FROM Waypoint ID
+/// 7. Checksum
+
+#[derive(Debug, PartialEq)]
+pub struct BwwData {
+    pub true_bearing: Option<f32>,
+    pub magnetic_bearing: Option<f32>,
+    pub to_waypoint_id: Option<ArrayString<MAX_LEN>>,
+    pub from_waypoint_id: Option<ArrayString<MAX_LEN>>,
+}
+
+fn do_parse_bww(i: &str) -> Result<BwwData, Error> {
+    // 1. Bearing, degrees True
+    let (i, true_bearing) = opt(float)(i)?;
+    let (i, _) = char(',')(i)?;
+    // 2. T = True
+    let (i, _) = opt(char('T'))(i)?;
+    let (i, _) = char(',')(i)?;
+
+    // 3. Bearing, degrees Magnetic
+    let (i, magnetic_bearing) = opt(float)(i)?;
+    let (i, _) = char(',')(i)?;
+    // 4. M = Magnetic
+    let (i, _) = opt(char('M'))(i)?;
+    let (i, _) = char(',')(i)?;
+
+    // 5. TO Waypoint ID
+    let (i, to_waypoint_id) = opt(is_not(","))(i)?;
+
+    let to_waypoint_id = if let Some(to_waypoint_id) = to_waypoint_id {
+        Some(
+            ArrayString::from(to_waypoint_id)
+                .map_err(|_e| Error::SentenceLength(to_waypoint_id.len()))?,
+        )
+    } else {
+        None
+    };
+
+    // 6. FROM Waypoint ID
+    let (i, _) = char(',')(i)?;
+    let (_i, from_waypoint_id) = opt(is_not(",*"))(i)?;
+
+    let from_waypoint_id = if let Some(from_waypoint_id) = from_waypoint_id {
+        Some(
+            ArrayString::from(from_waypoint_id)
+                .map_err(|_e| Error::SentenceLength(from_waypoint_id.len()))?,
+        )
+    } else {
+        None
+    };
+
+    Ok(BwwData {
+        true_bearing,
+        magnetic_bearing,
+        to_waypoint_id,
+        from_waypoint_id,
+    })
+}
+
+/// # Parse BWW message
+///
+/// See: <https://gpsd.gitlab.io/gpsd/NMEA.html#_bww_bearing_waypoint_to_waypoint>
+pub fn parse_bww(sentence: NmeaSentence) -> Result<BwwData, Error> {
+    if sentence.message_id != SentenceType::BWW {
+        Err(Error::WrongSentenceHeader {
+            expected: SentenceType::BWW,
+            found: sentence.message_id,
+        })
+    } else {
+        Ok(do_parse_bww(sentence.data)?)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use approx::assert_relative_eq;
+
+    use super::*;
+    use crate::parse::parse_nmea_sentence;
+
+    #[test]
+    fn test_parse_bww_full() {
+        let sentence = parse_nmea_sentence("$GPBWW,213.8,T,218.0,M,TOWPT,FROMWPT*42").unwrap();
+        assert_eq!(sentence.checksum, sentence.calc_checksum());
+        assert_eq!(sentence.checksum, 0x42);
+
+        let data = parse_bww(sentence).unwrap();
+
+        assert_relative_eq!(data.true_bearing.unwrap(), 213.8);
+        assert_relative_eq!(data.magnetic_bearing.unwrap(), 218.0);
+        assert_eq!(&data.to_waypoint_id.unwrap(), "TOWPT");
+        assert_eq!(&data.from_waypoint_id.unwrap(), "FROMWPT");
+    }
+
+    #[test]
+    fn test_parse_bww_with_optional_fields() {
```

Comment:
I think we should strive to allow sentences parsing to be successful only when at least 1 parameter is present when all parameters are `Option`-al

- [ ] Would you mind applying this change?

Reply:
Is such message technically valid? If it is, then we should allow it, as technically it's currect. Might be useless, but valid. Also, maybe such message can be treated as information itself. As an indicator that something isn't working (just an idea from the top of my head).

Reply:
As we already discussed it - you're correct. Let's keep the empty messages, especially if they have valid structure and checksum.

We should be clear in the docs about that though.
We can update the docs in subsequent PR.